Packs/PrismaSASE/Integrations/PrismaSASE/PrismaSASE.py

```diff
@@ -0,0 +1,819 @@
+from CommonServerPython import *  # noqa # pylint: disable=unused-wildcard-import
+from CommonServerUserPython import *  # noqa
+
+import requests
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()  # pylint: disable=no-member
+
+# CONSTANTS
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+SEARCH_LIMIT = 200
+
+
+SECURITYRULE_FIELDS = {
+    "name": "",
+    "action": "",
+    "description": "",
+    "log_setting": "",
+    "application": [],
+    "category": [],
+    "destination": [],
+    "destination_hip": [],
+    "profile_setting": {},
+    "service": [],
+    "source": [],
+    "source_hip": [],
+    "source_user": [],
+    "tag": [],
+    "from": [],
+    "to": [],
+    "disabled": "",
+    "negate_source": "",
+    "negate_destination": ""
+}
+
+
+class Client(BaseClient):
+    """Client class to interact with the service API
+
+    This Client implements API calls to the Saas Security platform, and does not contain any XSOAR logic.
+    Handles the token retrieval.
+
+    :param base_url (str): Saas Security server url.
+    :param client_id (str): client ID.
+    :param client_secret (str): client secret.
+    :param verify (bool): specifies whether to verify the SSL certificate or not.
+    :param proxy (bool): specifies if to use XSOAR proxy settings.
+    """
+
+    def __init__(self, base_url: str, client_id: str, client_secret: str, oauth_url: str, verify: bool, proxy: bool, **kwargs):
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.oauth_url = oauth_url
+
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy, **kwargs)
+
+    @staticmethod
+    def build_security_rule(args: dict):
+
+        # This method will combine the args into a JSON formatted request body
+        rule = {}
+        keys = args.keys()
+        for key in SECURITYRULE_FIELDS:
+            if key in keys:
+
+                if args.get(key) != "" and args.get(key) != "None":
+                    if key == 'profile_setting':
+                        if isinstance(args.get(key), list):
+                            val = args.get(key)
+                        else:
+                            val = [x.strip() for x in args.get(key).split(',')]   # type: ignore
+
+                        rule[key] = {'group': val}
+
+                    elif isinstance(SECURITYRULE_FIELDS.get(key), str):
+                        rule[key] = args.get(key)   # type: ignore
+
+                    elif isinstance(SECURITYRULE_FIELDS.get(key), list):
+                        if isinstance(args.get(key), list):
+                            val = args.get(key)
+                        else:
+                            val = [x.strip() for x in args.get(key).split(';')]   # type: ignore
+                        rule[key] = val   # type: ignore
+
+        return rule
+
+    def create_security_rule(self, rule: dict, folder: str, position: str, tsg_id: str):
+        uri = '/sse/config/v1/security-rules'
```

Comment:
hey Shachar, this is true for the commands that I've added here.  However, the SASE API has several endpoints that would have a different URI prefix so I thought for future proofing to not make it a constant.  Maybe a good compromise is to have a constant for each of those prefixs and label them "CONFIG_URI" or something.  I'll do that now and let me know if you think there is a better way.

Reply:
OK, in that case, when we will add the additional commands we should add a constant for each relevant endpoint (if it will be used more than once in the code).